services/bettersocial/chat/allAnonChat.js

```diff
@@ -0,0 +1,129 @@
+const {User, ChatAnonUserInfo} = require('../../../databases/models');
+const {Op} = require('sequelize');
+const crypto = require('crypto');
+const BetterSocialConstantListUtils = require('../constantList/utils');
+
+const generate_channel_id_for_anon_chat = (owner, member) => {
+  const hash = crypto.createHash('sha256');
+  hash.update(owner, member);
+  return hash.digest('hex');
+};
+
+const is_all_anon_user = async (members) => {
+  // check all members users are anonymous
+  let users = await User.findAll({
+    where: {
+      user_id: {
+        [Op.in]: members
+      },
+      is_anonymous: false
+    }
+  });
+  return users.length === 0 ? true : false;
+};
+
+const generate_anon_user_info = () => {
+  const emoji = BetterSocialConstantListUtils.getRandomEmoji();
+  const color = BetterSocialConstantListUtils.getRandomColor();
+  return {
+    anon_user_info_color_code: color.code,
+    anon_user_info_color_name: color.color,
+    anon_user_info_emoji_code: emoji.emoji,
+    anon_user_info_emoji_name: emoji.name
+  };
+};
+
+const get_anon_info_from_old_channel = async (oldChannelId, member) => {
+  let anonInfo = {};
+  if (oldChannelId) {
+    let previous_channel = await ChatAnonUserInfo.findOne({
+      where: {
+        channel_id: oldChannelId,
+        my_anon_user_id: member
+      }
+    });
+    anonInfo = {
+      anon_user_info_color_code: previous_channel.anon_user_info_color_code,
+      anon_user_info_color_name: previous_channel.anon_user_info_color_name,
+      anon_user_info_emoji_code: previous_channel.anon_user_info_emoji_code,
+      anon_user_info_emoji_name: previous_channel.anon_user_info_emoji_name
+    };
+  } else {
+    anonInfo = generate_anon_user_info();
+  }
+  return anonInfo;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `get_anon_info_from_old_channel` function, you are not handling the case where `previous_channel` is null. If `ChatAnonUserInfo.findOne` does not find a match, it will return null and accessing properties on `previous_channel` will throw an error. You should add a null check before accessing its properties.

```diff
- anonInfo = {
-   anon_user_info_color_code: previous_channel.anon_user_info_color_code,
-   anon_user_info_color_name: previous_channel.anon_user_info_color_name,
-   anon_user_info_emoji_code: previous_channel.anon_user_info_emoji_code,
-   anon_user_info_emoji_name: previous_channel.anon_user_info_emoji_name
- };
+ if (previous_channel) {
+   anonInfo = {
+     anon_user_info_color_code: previous_channel.anon_user_info_color_code,
+     anon_user_info_color_name: previous_channel.anon_user_info_color_name,
+     anon_user_info_emoji_code: previous_channel.anon_user_info_emoji_code,
+     anon_user_info_emoji_name: previous_channel.anon_user_info_emoji_name
+   };
+ } else {
+   anonInfo = generate_anon_user_info();
+ }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->